src/main/java/mate/academy/bookshop/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+    private final UserRepository userRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, PlaceOrderRequestDto requestDto) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "There is not found user with id " + userId));
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                "There is not found shopping cart for user with id " + userId));
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.PENDING);
+        Set<OrderItem> orderItems = new HashSet<>();
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        if (cartItems.isEmpty()) {
+            throw new EntityNotFoundException(
+                    "There is not found any cart items in the shopping cart of user with id "
+                            + userId);
+        }
+        BigDecimal total = BigDecimal.ZERO;
+        for (CartItem cartItem: cartItems) {
+            Book book = cartItem.getBook();
+            int quantity = cartItem.getQuantity();
+            BigDecimal price = cartItem.getBook().getPrice();
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+            orderItem.setBook(book);
+            orderItem.setPrice(price);
+            orderItem.setQuantity(quantity);
+            orderItems.add(orderItem);
+            total.add(price.multiply(BigDecimal.valueOf(quantity)));
+            cartItem.setDeleted(true);
+        }
+        order.setTotal(total);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setOrderItems(orderItems);
+        Order savedOrder = orderRepository.save(order);
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    public List<OrderDto> getAll(Long userId, Pageable pageable) {
+        return orderRepository.findAllByUserId(userId).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
+    public OrderDto updateStatusOrder(Long orderId,
+                                          UpdateStatusRequestDto statusRequestDto) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "There is not found order with id " + orderId));
+        order.setStatus(statusRequestDto.getStatus());
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public Set<OrderItemDto> getAllOrderItems(Long userId, Long orderId, Pageable pageable) {
```

Comment:
Improve this method the same way you did in `getItemById`. (Create similar custom repository method but without specification of `orderItemId`)